### DISCUSSION TITLE
GeometryCollection in GeoJSON

### DISCUSSION THREAD

**Discussion by TextTL** [visgl/deck.gl#7146] [Q&A] [answered]
Hi,

I'm using GeometryCollection to compose symbols, and I couldn't figure out how to get the index of geometry in the collection so that I can pass the correct color for the geometry; for example, I got a Feature as below:

 {
            type: "Feature",
            geometry: {
                type: "GeometryCollection",
                geometries: [
                    {
                        type: "Point",
                        coordinates: [0,0]
                    },
                    {
                        type: "Point,
                        coordinates: [10,0 ]
                    }
                    ]
            },
            properties: {
             color: [[0, 255, 255], [255,0,255] ]
             radius: [10,5]
            },
}

I want to have different colors and radii for different circles. The getFillColor and getPointRadius accessors pass the Feature, and I couldn't figure out which geometry is being rendered to provide desired colors and radius. Do you know how to achieve that? can I extend GeoJSONLayer to pass the index of the geometry as an optional parameter? This would be very helpful and make GeoJSON to render more complex compositions of shapes.

Thank you!

**Comment by Pessimistress** [comment, marked as answer]
Accessors are called per feature. If you want different colors for each geometry then you need to break them into separate features.

**Reply by TextTL** [reply]
I had that, and I have no problem with the display using Feature for each geometry. And then I have a use case to drag or copy the symbol, and I ran into the hardship group them together to drag/copy Features together. I found GeometryCollection could satisfy the use case because pickinfo returns the Feature, which I have all geometries to do transformation with ease. and I ran into problems with displaying individual geometries. Before creating CompositeLayers for symbols that I need, I think it could be useful if the accessor passes the index of the geometry so that the accessor can provide desired values for the geometry instead of creating CompositeLayer.

Thank you for your answer!
